plugins/vswitch/vswitch.cpp

```diff
@@ -609,6 +611,67 @@ class wf_vswitch_global_plugin_t : public wf::per_output_plugin_t<vswitch>
 
         return wf::ipc::json_ok();
     };
+
+    wf::ipc::method_callback send_view = [=] (const wf::json_t& data)
+    {
+        uint64_t x = wf::ipc::json_get_uint64(data, "x");
+        uint64_t y = wf::ipc::json_get_uint64(data, "y");
+        uint64_t output_id = wf::ipc::json_get_uint64(data, "output-id");
+        std::optional<uint64_t> view_id = wf::ipc::json_get_optional_uint64(data, "view-id");
+
+        auto wo = wf::ipc::find_output_by_id(output_id);
+        if (!wo)
+        {
+            return wf::ipc::json_error("Invalid output!");
+        }
+
+        auto grid_size = wo->wset()->get_workspace_grid_size();
+        if ((int(data["x"]) >= grid_size.width) || (int(data["y"]) >= grid_size.height))
+        {
+            return wf::ipc::json_error("Workspace coordinates are too big!");
+        }
+
+        wayfire_toplevel_view switch_with_view;
+        if (!view_id.has_value())
+        {
+            return wf::ipc::json_error("view-id not specified!");
+        }
+
+        auto view = toplevel_cast(wf::ipc::find_view_by_id(view_id.value()));
+        if (!view)
+        {
+            return wf::ipc::json_error("Invalid view or view not toplevel!");
+        }
+
+        if (!view->is_mapped())
+        {
+            return wf::ipc::json_error("Cannot grab unmapped view!");
+        }
+
+        if (view->get_output() != wo)
+        {
+            return wf::ipc::json_error("Cannot grab view on a different output!");
+        }
+
+        wf::view_change_workspace_signal signal;
```

Comment:
I believe that pretty much the same code has been implemented already in the same file, maybe we can have a shared function instead of duplicating the code?